common/workunit/workunit.cpp

```diff
@@ -6959,23 +6928,42 @@ int verifyWorkunitDAToken(const char * distributedAccessToken)
         }
     }
 
+    //Isolate string between []
+    StringBuffer tokenValues(distributedAccessToken + 5);
```

Comment:
This could be common code with above.  with options to retrieve w= and/or u=.

Reply:
The common code could just populate the StringArray, and/or extract the parameters directly.